src/Workspaces/CSharp/Portable/CodeStyle/CSharpCodeStyleOptions.cs

```diff
@@ -156,11 +159,14 @@ private static Option<T> CreateOption<T>(OptionGroup group, string name, T defau
                     GetExpressionBodyPreferenceEditorConfigString),
                 new RoamingProfileStorageLocation($"TextEditor.CSharp.Specific.{nameof(PreferExpressionBodiedLambdas)}")});
 
-        public static readonly Option<CodeStyleOption<bool>> PreferBraces = CreateOption(
+        public static readonly Option<CodeStyleOption<PreferBracesPreference>> PreferBraces = CreateOption(
             CSharpCodeStyleOptionGroups.CodeBlockPreferences, nameof(PreferBraces),
-            defaultValue: CodeStyleOptions.TrueWithSilentEnforcement,
+            defaultValue: UseBracesWithSilentEnforcement,
             storageLocations: new OptionStorageLocation[] {
-                EditorConfigStorageLocation.ForBoolCodeStyleOption("csharp_prefer_braces"),
+                new EditorConfigStorageLocation<CodeStyleOption<PreferBracesPreference>>(
+                    "csharp_prefer_braces",
+                    s => ParsePreferBracesPreference(s, UseBracesWithSilentEnforcement),
+                    GetPreferBracesPreferenceEditorConfigString),
                 new RoamingProfileStorageLocation($"TextEditor.CSharp.Specific.{nameof(PreferBraces)}")});
```

Comment:
@Neme12 Do you remember how serialization compatibility is preserved for roaming settings? The previous true/false values are still supported, but have new names.

Reply:
The relevant code is around here: https://github.com/dotnet/roslyn/blob/8442a69c93a4a5d76dce7d28c2e0bb3d35c149c7/src/VisualStudio/Core/Def/Implementation/Options/RoamingVisualStudioProfileOptionPersister.cs#L139-L145

My recollections is that we have code somewhere to go from bools to enums.  But the assumption is that the false/true values of the bools should map to 0/1.  That's how we handled "expression-body" preferences in the past i believe.  But my memory is hazy.

Reply:
It's actually the last two lines of this method:

https://github.com/dotnet/roslyn/blob/61bd5ed8a64c049280f000cba60831cb6a4b15fd/src/Workspaces/Core/Portable/CodeStyle/CodeStyleOption.cs#L159-L169

I've updated the enum to reflect this and document the requirement.

Reply:
@sharwell In my PR for naming styles, the value was a string which contained serialized XML that had its own information about the version of the format. I don't think that is applicable here.

Reply:
If this ends up being just a change of the type after all, what happens if an enum value  of 2 is read by an older version that expects a bool?

Reply:
To avoid any problems, we could reuse the same approach I did back then to make sure that newer version find the value in the old storage location but only ever save it to a new location so that older versions are not affected by this.

There would be a new RoamingProfileStorageLocation added to the list of storage locations:
https://github.com/dotnet/roslyn/blob/10ffcf763eb85e4eb1fe3d413b6d0ccdcd6bb8f8/src/Workspaces/Core/Portable/Simplification/SimplificationOptions.cs#L103-L108

Reply:
> If this ends up being just a change of the type after all, what happens if an enum value of 2 is read by an older version that expects a bool?

This shouldn't be a problem since the new serialization is for the first release of Visual Studio 2019, and those settings shouldn't propagate backward to interfere with roaming settings of Visual Studio 2017. I'm still working on verifying this.